### PR TITLE
Add more teambuilder support for Let's Go

### PR DESCRIPTION
**Pull request by KrisXV**
@Zarel

### CODE REVIEW

**Comment by TheImmortal** [comment]
Change the tiers to Uber, OU, NFE, LC.

**Comment by TheImmortal** [comment]
And if you can edit the mod on the server as well to change the tiers to those then that'll be cool. Basically, it's a new game so everything starts at OU. Mewtwo is the only Uber for now.

**Comment by Zarel** [comment]
So, none of this is your fault, but the code in this file is really bad and needs to be redone at some point.

File: js/client-teambuilder.js [hunk 1]
```diff
@@ -1069,6 +1069,7 @@
 		},
 		renderSet: function (set, i) {
 			var template = Tools.getTemplate(set.species);
+			var isNotLetsGo = !this.curTeam.format.startsWith('gen7letsgo');
```
Comment (Zarel) [inline js/client-teambuilder.js]: I don't like the double negative here. Just do `isLetsGo`

**Comment by KrisXV** [comment]
done @Zarel

File: js/search.js [hunk 2]
```diff
@@ -846,6 +849,8 @@
 			else if (format === 'doublesou') tierSet = tierSet.slice(slices.DOU);
 			else if (format === 'doublesuu') tierSet = tierSet.slice(slices.DUU);
 			else if (format === 'doublesnu') tierSet = tierSet.slice(slices.DNU);
+			else if (isLetsGo && format.includes('ou')) tierSet = tierSet.slice(slices.OU);
```
Comment (Zarel) [inline js/search.js]: Why are you doing `includes`? Shouldn't you just strip off the preamble so you're actually just left with `ou`?

File: js/search.js [hunk 3]
```diff
@@ -793,6 +793,7 @@
 		var requirePentagon = (format === 'battlespotsingles' || format === 'battledoubles' || format.slice(0, 3) === 'vgc');
 		var template;
 		var isBH = (format === 'balancedhackmons' || format === 'bh');
+		var isLetsGo = format.startsWith('letsgo');
```
Comment (Zarel) [inline js/search.js]:
```suggestion
```

File: js/search.js [hunk 4]
```diff
@@ -793,6 +793,7 @@
 		var requirePentagon = (format === 'battlespotsingles' || format === 'battledoubles' || format.slice(0, 3) === 'vgc');
```
Comment (Zarel) [inline js/search.js]:
```suggestion
		var isLetsGo = format.startsWith('letsgo');
		if (isLetsGo) format = format.slice(6);
		var requirePentagon = (format === 'battlespotsingles' || format === 'battledoubles' || format.slice(0, 3) === 'vgc');
```

File: js/search.js (hunk 2 above)
Comment (Zarel) [reply js/search.js]:
```suggestion
```

File: build-tools/build-indexes [hunk 5]
```diff
@@ -301,6 +302,17 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 				if (template.tier === 'CAP' || template.tier === 'CAP NFE' || template.tier === 'CAP LC') {
 					return template.tier;
 				}
+				if (isLetsGo) {
+					let baseTemplate = Tools.mod(gen).getTemplate(template.baseSpecies);
+					if ((baseTemplate.num > 151 || baseTemplate.num < 1) && baseTemplate.num !== 808 && baseTemplate.num !== 809) {
+						if (!['Alola', 'Mega', 'Mega-X', 'Mega-Y', 'Starter'].includes(template.forme)) return 'Illegal';
```
Comment (Zarel) [inline build-tools/build-indexes]: err, this line does nothing

Comment (Zarel) [reply build-tools/build-indexes]: I'm pretty sure you're making the same mistake you made server-side.

Comment (Zarel) [reply build-tools/build-indexes]: https://github.com/Zarel/Pokemon-Showdown/commit/ed54a6eb4db5d5271174cc42537432bb0bf1e602

You understand how this change made the logic a lot clearer, and makes it a lot harder to make mistakes, right?

Comment (Zarel) [reply build-tools/build-indexes]: Can you do the same kind of change here?

Comment (KrisXV) [reply build-tools/build-indexes]: done

**Pull request merged**